src/components/Comments/Comment.js

```diff
@@ -167,6 +172,19 @@ const Comment = ({
     }
   };
 
+  const onPressDm = async () => {
+    try {
+      refSheet.current.open();
+      setLoading({...loading, loadingGetAllowAnonDmStatus: true});
+      const data = await getAllowAnonDmStatus('comment', comment?.id);
+      setUserAllowDm(data?.user.allow_anon_dm);
+    } catch (e) {
+      console.log(e);
+    } finally {
+      setLoading({...loading, loadingGetAllowAnonDmStatus: false});
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

A new function `onPressDm` has been added to handle the press event on the DM button. It fetches the status of whether the user allows anonymous DMs or not. Make sure error handling is properly done for network errors or server-side issues.

```javascript
const onPressDm = async () => {
  try {
    refSheet.current.open();
    setLoading({...loading, loadingGetAllowAnonDmStatus: true});
    const data = await getAllowAnonDmStatus('comment', comment?.id);
    setUserAllowDm(data?.user.allow_anon_dm);
  } catch (e) {
    console.log(e);
  } finally {
    setLoading({...loading, loadingGetAllowAnonDmStatus: false});
  }
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->